sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StateTypesEncoderUtils.scala

```diff
@@ -62,33 +70,79 @@ class StateTypesEncoder[GK, V](
   private val rowToObjDeserializer = valExpressionEnc.resolveAndBind().createDeserializer()
   private val reusedValRow = new UnsafeRow(valEncoder.schema.fields.length)
 
+  private val NO_TTL_ENCODED_VALUE: Long = -1L
```

Comment:
Could we avoid recording this explicitly ?

Reply:
Done.